### PR TITLE
Sidebar Component

### PR DESCRIPTION
**Pull request by hassanelnajjar**
# Sidebar Component
- [x] Add Sidebar component from AntDesign
- [x] Add props (Content Component - Logout Function)
- [x] Add layout for the whole page with side bar and content.

#52

### CODE REVIEW

File: client/src/index.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+const TodaySchedule = () => <h3>Today Schedule</h3>;
+const Patients = () => <h3>Patients</h3>;
+const Calender = () => <h3>Calender</h3>;
+
+const options = [<TodaySchedule />, <Patients />, <Calender />];
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Sidebar contentComponents={options} />
```
Comment (Mu7ammadAbed) [inline client/src/index.js:16]: You should require this container at the `App.js` when we are at the `/dashboard` route

File: client/src/components/Sidebar/index.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  const [itemKey, setItemKey] = useState('1');
+
+  const onCollapse = (collapsedValue) => setCollapsed(collapsedValue);
+
+  return (
+    <Layout className="page-layout">
+      <Sider collapsible collapsed={collapsed} onCollapse={onCollapse}>
+        <div className={collapsed ? 'logo logo-hidden' : 'logo'}>
+          <div>
+            <img src={logoImage} alt="" />
```
Comment (Mu7ammadAbed) [inline client/src/components/Sidebar/index.js:27]: Convert this into a common component at `/components/common/Image` and pass the `src` and `alt` to it as a props

File: client/src/components/Sidebar/index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          </div>
+          <h3 className="title-text">Dental Clinic</h3>
```
Comment (Mu7ammadAbed) [inline client/src/components/Sidebar/index.js:29]: Use the `Title` component from the `antd`

Comment (hassanelnajjar) [reply client/src/components/Sidebar/index.js]: Done

File: client/src/components/Sidebar/index.js (hunk 2 above)
Comment (hassanelnajjar) [reply client/src/components/Sidebar/index.js]: Done

File: client/src/index.js (hunk 1 above)
Comment (hassanelnajjar) [reply client/src/index.js]: Done

File: client/src/components/Sidebar/index.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            />
+          </nav>
+        </Header>
+        <Content className="site-layout-content">
+          <div className="site-layout-content-body">
+            {contentComponents[+itemKey - 1]}
+          </div>
+        </Content>
+        <Footer className="site-layout-footer">
+          Dental Clinic ©2021 Created by dentoro
```
Comment (Mu7ammadAbed) [inline client/src/components/Sidebar/index.js:84]:
```suggestion
          Dental Clinic &copy;2021 Created by Dentoro
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: client/src/components/common/LogoImage/index.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { string } from 'prop-types';
+
+function LogoImage({ src, alt }) {
+  return (
+    <div>
+      <img src={src} alt={alt} />
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (hassanelnajjar) [reply client/src/components/common/LogoImage/index.js:10]: Done

File: client/src/components/common/LogoImage/index.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+LogoImage.propTypes = {
```
[the remark this answers is not in the document]
Comment (hassanelnajjar) [reply client/src/components/common/LogoImage/index.js:12]: Done

File: client/src/App/index.js [hunk 7]
```diff
@@ -1,25 +1,25 @@
...
+import Sidebar from '../components/Sidebar';
+
+const TodaySchedule = () => <h3>Today Schedule</h3>;
+const Patients = () => <h3>Patients</h3>;
+const Calender = () => <h3>Calender</h3>;
+
+const options = [<TodaySchedule />, <Patients />, <Calender />];
+
 function App() {
   return (
```
[the remark this answers is not in the document]
Comment (hassanelnajjar) [reply client/src/App/index.js:14]: Done

File: client/src/components/Sidebar/index.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  LogoutOutlined,
+  MenuFoldOutlined,
+} from '@ant-design/icons';
+import { arrayOf, element } from 'prop-types';
+import logo from '../../assets/images/logo.png';
+import LogoImage from '../common/LogoImage';
+
+const { Header, Content, Footer, Sider } = Layout;
+const { Title } = Typography;
+function Sidebar({ contentComponents }) {
```
[the remark this answers is not in the document]
Comment (hassanelnajjar) [reply client/src/components/Sidebar/index.js:20]: Done

File: client/src/App/index.js [hunk 9]
```diff
@@ -1,25 +1,25 @@
...
-        </a>
-      </header>
-    </div>
+    <Switch>
+      <Route exact path="/">
+        Home
+      </Route>
+      <Route exact path="/dashboard">
+        <Sidebar contentComponents={options} />;
+      </Route>
```
[the remark this answers is not in the document]
Comment (hassanelnajjar) [reply client/src/App/index.js:21]: Done

File: client/src/components/Sidebar/index.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+  const [collapsed, setCollapsed] = useState(false);
+  const [itemKey, setItemKey] = useState('1');
+  const history = useHistory();
+
+  const onCollapse = (collapsedValue) => setCollapsed(collapsedValue);
+  const logoutFunction = () => history.push('/');
```
[the remark this answers is not in the document]
Comment (hassanelnajjar) [reply client/src/components/Sidebar/index.js:26]: Done

File: client/src/components/Sidebar/index.js (hunk 4 above)
Comment (hassanelnajjar) [reply client/src/components/Sidebar/index.js:84]: Done

File: client/src/components/common/Image/index.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { string } from 'prop-types';
+
+const Image = ({ src, alt }) => (
+  <div>
+    <img src={src} alt={alt} />
+  </div>
```
Comment (Mu7ammadAbed) [inline client/src/components/common/Image/index.js:7]:
```suggestion
    <img src={src} alt={alt} />
```

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Alaalser — APPROVED** [review]

**Review by AlaaShurrab — APPROVED** [review]

**Pull request merged**